azavea/pfb-network-connectivity
# Add tilegarden

## Overview

Copies over enough of [Tilegarden](https://github.com/azavea/tilegarden) to get a tiler service running in development and serving census blocks, ways, and bike infrastructure tiles (though not filtered by job).

It doesn't copy over any of the configuration code (i.e. [`.env.template`](https://github.com/azavea/tilegarden/blob/develop/.env.template) and the [terraform config](https://github.com/azavea/tilegarden/tree/develop/src/terraform)), since the plan is to integrate at least some of that into the existing deployment infrastructure for this project.  So it's development-only at the moment.  There are a few remnants (e.g. `deploy` and related commands in the `project.json` file) that are in what was copied but won't currently work.

Still to do:
- Add filtering by job (#609)
- Modify the tile URLs returned by the API (#592)
- Get it deploying (i.e. adapt and integrate Tilegarden's deployment process into the scripts and config for this project) (#593, #594, #595)

### Demo

![image](https://user-images.githubusercontent.com/6598836/49315023-485dc780-f4ba-11e8-9a22-32d7edc7e334.png)

### Notes

See the commit messages for some additional notes and details.

## Testing Instructions

If you haven't run the migration from PR #600, run that first.  If you don't have any completed analysis jobs at all, you'll need to run one to have some data to look at.

- Run `scripts/setup` or (in the VM) `scripts/update` to build, then `scripts/server`
- Go to the front-end app (http://localhost:9301), click through to a neighborhood, and show one of the layers listed under "Overlays" in the layer picker.
- Open up the debug console and pick the "Network" tab, then move or zoom the map to get it to request some new tiles.
- Copy the URL of one of the tile requests from your development S3 tile source, then replace everything up to the first number (i.e. the Z value in the tile path) with http://localhost:9400/tile/, e.g. http://localhost:9400/tile/14/4770/6200.png.
- That request will fail, telling you you need to supply a config parameter.  Add `?config=ways`, `?config=census_blocks`, or `?config=bike_infrastructure` to see some actual tiles.

Also, `scripts/test` should run Tilegarden's tests in addition to the Django tests, and should pass.

Resolves #591.
